annis-service/src/main/java/annis/sqlgen/SolutionSqlGenerator.java

```diff
@@ -217,12 +245,21 @@ else if(order == OrderType.ascending)
         ids.add("sourceIdx");
       }
       // add the node ID for each output node an the category ID
-      for (int i = 1; i <= queryData.getMaxWidth(); ++i)
+      int i = 0;
+      for (QueryNode node : alternative)
```

Comment:
Replacing the `i <= queryData.getMaxWidth()` condition is problematic for OR-queries without variables. E.g.
```
tok=‎"zu‎" | tok=‎/ab.*‎/
```
would lead to the invalid order clause
```
ORDER BY id1 ASC, cat1 ASC, id2 ASC, cat2 ASC
```
with this change (there is no `id2` or `cat2`).

The same is valid for the MatrixSqlGenerator before. I still don't understand why the `SELECT` clause does not have the same problem.

Reply:
This problem is caused by passing the nodes from all alternatives in AbstractUnionSql generator:
https://github.com/korpling/ANNIS/blob/9deed30b0b5eb1fecc695f224864956823209107/annis-service/src/main/java/annis/sqlgen/AbstractUnionSqlGenerator.java#L57-L59

I have changed it to take only one set of alternatives with maxWidth as in the AbstractSqlGenerator:
https://github.com/korpling/ANNIS/blob/4de11bc414e8dc1e4dd236c321444448cf0a1f99/annis-service/src/main/java/annis/sqlgen/AbstractSqlGenerator.java#L73-L78

Now your example query works.